src/pages/index.js

```diff
@@ -4,21 +4,68 @@ import Section from "../components/Section.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import UserInfo from "../components/UserInfo.js";
-import { initialCards, formValidationOptions } from "../utils/constants.js";
+import { formValidationOptions, initialCards } from "../utils/constants.js";
+import Api from "../utils/Api.js";
 import "./index.css";
 
+// Initialize the API
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "6b68a929-d08e-4cf9-a697-37317d009dd7",
+    "Content-Type": "application/json",
+  },
+});
+
 function handleImageClick(name, link) {
   popupWithImage.open({ name, link });
 }
 
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
   return card.generateCard();
 }
 
+function handleDeleteClick(card) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitAction(() => {
+    return api
+      .deleteCard(card._id)
+      .then(() => {
+        card._deleteCard();
+        deleteCardPopup.close();
+      })
+      .catch((err) => {
+        console.error("Failed to delete card:", err);
+      });
+  });
+}
+
+function handleLikeClick(card) {
+  const likeAction = card._isLiked
```

Comment:
`_isLiked`

* Please, delete the **low dash** (`underscore`) from methods if you want to use them `outside` a class because the `low dash` shows a **private** method which is used only inside the class.